node/core/approval-voting/src/lib.rs

```diff
@@ -1982,7 +2006,7 @@ async fn launch_approval(
 					(candidate_hash, candidate.descriptor.para_id),
 				);
 				// do nothing. we'll just be a no-show and that'll cause others to rise up.
-				metrics.on_approval_unavailable();
+				metrics_guard.forget_with(|m| m.on_approval_unavailable());
```

Comment:
Nit: I would be more in favor of creating named functions compared to a closure. That would complicate the definition but will make the call sites cleaner which is much more important IMO.